public/components/metrics/sidebar/metric_name.tsx

```diff
@@ -11,10 +11,15 @@ import { OBSERVABILITY_CUSTOM_METRIC } from '../../../../common/constants/metric
 
 const MetricIcon = ({ metric }) => {
   const metricIcons = useSelector(metricIconsSelector);
-  const iconMeta = metricIcons[metric.catalog];
-  if (metric.catalog === OBSERVABILITY_CUSTOM_METRIC)
+  const iconMeta = metricIcons[metric?.catalog];
+  // console.log('iconMeta: ', iconMeta);
+  const metricCatalog = metric?.catalog;
+  // console.log('metricCatalog: ', metricCatalog);
+  if (metric?.catalog === OBSERVABILITY_CUSTOM_METRIC || metric?.catalog === 'OpenTelemetry') {
```

Comment:
use a variable for `OpenTelemetry` instead of hard coded string since `metric?.catalog` doesn't have type checks

Reply:
changed it